### PR TITLE
Added sparse GSI support

### PR DESCRIPTION
**Pull request by samritchie**
Addresses #3. This PR modifies `KeyAttributeSchema` validation to allow wrapper (`Nullable`/`Option`) attribute types, and changes the `UpdateExpr` check to allow modification of non-primary key attributes. I even added tests! 🎉 

@eiriktsarpalis I’d appreciate a once-over/feedback if you have time.

Thanks!

### CODE REVIEW

**Comment by eiriktsarpalis** [comment]
Wow, great work. 

I reviewed the code (poorly, just had a look at the diff) and I think I have an identified an issue where supplying an optional HashKey in the presence of a GlobalSecondaryIndex would not result in a error, which it should. If that is just a case of me not understanding the code I wrote one year ago, could you please also add a test the checks for this?

File: src/FSharp.AWS.DynamoDB/RecordKeySchema.fs [hunk 1]
```diff
@@ -138,7 +141,7 @@ type RecordTableInfo with
     static member FromRecordPickler<'T> (pickler : RecordPickler<'T>) =
         let hkcaOpt = typeof<'T>.TryGetAttribute<ConstantHashKeyAttribute> ()
         let rkcaOpt = typeof<'T>.TryGetAttribute<ConstantRangeKeyAttribute> ()
-        let mkKAS rp = KeyAttributeSchema.Create rp
+        let mkKAS rp = KeyAttributeSchema.Create(rp, true)
```
Comment (eiriktsarpalis) [inline src/FSharp.AWS.DynamoDB/RecordKeySchema.fs]: I think it relates to how this declaration is being used further down

Comment (eiriktsarpalis) [inline src/FSharp.AWS.DynamoDB/RecordKeySchema.fs]: It's being applied to the HashKey in lines 216-219

**Comment by samritchie** [comment]
Thanks @eiriktsarpalis - I added a test. As far as I can tell, `mkKAS` is only called for GSI Hash Keys - the primary hashes create the schema via `setResult` on line 169

**Comment by eiriktsarpalis** [comment]
Thanks. Merged and pushed as version 0.7.0

**Pull request merged**